tests/multi_material/run_benchmark.py

```diff
@@ -207,52 +200,50 @@ def write_output(output_file):
 
 if simulation.dimensional:
     approximation_parameters["g"] = simulation.g
-    approximation_parameters["T0"] = 0
+    approximation_parameters["T0"] = 0.0
 
-Ra = getattr(simulation, "Ra", 0)
+Ra = getattr(simulation, "Ra", 0.0)
 approximation = ga.BoussinesqApproximation(Ra, **approximation_parameters)
 
 # Timestep object
 real_func_space = fd.FunctionSpace(mesh, "R", 0)
 timestep = fd.Function(real_func_space).assign(simulation.initial_timestep)
 
-# Set up energy and Stokes solvers
-energy_solver = ga.EnergySolver(
-    temperature,
-    velocity,
-    approximation,
-    timestep,
-    ga.ImplicitMidpoint,
-    bcs=simulation.temp_bcs,
-)
-stokes_nullspace = ga.create_stokes_nullspace(
-    stokes_function.function_space(), **simulation.stokes_nullspace_args
-)
+# Set up possible energy solver
+energy_solver = None
+if hasattr(simulation, "initialise_temperature"):
+    energy_solver = ga.EnergySolver(
+        temperature,
+        velocity,
+        approximation,
+        timestep,
+        ga.ImplicitMidpoint,
+        bcs=simulation.temp_bcs,
+    )
+# Set up Stokes solver
+stokes_nullspace = ga.create_stokes_nullspace(stokes_function.function_space())
 stokes_solver = ga.StokesSolver(
     stokes_function,
     temperature,
     approximation,
     bcs=simulation.stokes_bcs,
-    quad_degree=None,
-    solver_parameters=simulation.stokes_solver_params,
     nullspace=stokes_nullspace,
     transpose_nullspace=stokes_nullspace,
 )
-
-# Solve initial Stokes system
-stokes_solver.solve()
+if benchmark == "schmalholz_2011":
+    stokes_solver.solver_parameters["snes_linesearch_type"] = "cp"
```

Comment:
Why is this needed? Add a comment so that if we return to this at a later time, we understand.